src/clients/oauth.rs

```diff
@@ -701,17 +701,29 @@ pub trait OAuthClient: BaseClient {
     ///
     /// Parameters:
     /// - limit - the number of entities to return
+    /// - time_limit - a Unix timestamp in milliseconds. Returns all items after
+    /// or before (but not including) this cursor position.
     ///
-    /// [Reference](https://developer.spotify.com/documentation/web-api/reference/#endpoint-get-the-users-currently-playing-track)
+    /// [Reference](https://developer.spotify.com/documentation/web-api/reference/#endpoint-get-recently-played)
     async fn current_user_recently_played(
         &self,
         limit: Option<u32>,
+        time_limit: Option<TimeLimits>,
     ) -> ClientResult<CursorBasedPage<PlayHistory>> {
         let limit = limit.map(|x| x.to_string());
-        let params = build_map! {
+        let mut params = build_map! {
             optional "limit": limit.as_deref(),
         };
 
+        let time_limit = match time_limit {
+            Some(TimeLimits::Before(y)) => Some(("before", y.timestamp().to_string())),
+            Some(TimeLimits::After(y)) => Some(("after", y.timestamp().to_string())),
```

Comment:
According to the document:
> `after`: A Unix timestamp in **milliseconds**
> `before`: A Unix timestamp in **milliseconds**

Then `DateTime:: timestamp` returns the number of non-leap **seconds** since January 1, 1970 0:00:00 UTC (aka "UNIX timestamp"). You should get the `milliseconds` value instead of `seconds` value.

Reply:
Good catch! My bad